tests/testthat/test-datamart-secrets.R

```diff
@@ -0,0 +1,203 @@
+localGlobal("companySecret", "test_company_secret")
+localGlobal("projectSecret", "test_project_secret")
+
+test_env = new.env() # holds HTTP header verification result that is put there by mocked HTTP function and is used by the tests below
+
+verifyHttpHeaders <- function(headers = list(), expect_company_secret_header_to_be_equivalent_to_company_secret, expect_project_secret_header_to_be_equivalent_to_project_secret) 
+{
+    companySecretHeader = headers["X-Q-Company-Secret"];
+    companySecretHeader <- ifelse(is.null(companySecretHeader), "", companySecretHeader)
+    projectSecretHeader = headers["X-Q-Project-Secret"];
+    projectSecretHeader <- ifelse(is.null(projectSecretHeader), "", projectSecretHeader)
```

Comment:
Some more null coalescing will make this easier to read in my opinion.

```suggestion
    companySecretHeader <- headers["X-Q-Company-Secret"] %||% ""
    projectSecretHeader <- headers["X-Q-Project-Secret"] %||% ""
```
minor nit: `;` not required in R and generally not used as end of line char.

Reply:
> Some more null coalescing will make this easier to read in my opinion.

I agree.
I also checked what `headers` actually is in terms of type/structure. Running `str(headers)` gives this
```
Named chr [1:3] "some_token" "some_project_token" "1027046"
 - attr(*, "names")= chr [1:3] "X-Q-Company-Secret" "X-Q-Project-Secret" "X-Q-Project-ID"
```
Copilot says to use double brackets to achieve what I want, i.e. `companySecretHeader = headers[["X-Q-Company-Secret"]] %||% ""`, and to change the default value of `headers` to `character(0)`.

> minor nit: ; not required in R and generally not used as end of line char.

Oh, yes. Thanks. This is my muscle memory from 3 other languages that all require `;`

Reply:
Yeah if headers is a list then it will return a list if using a single `[]` pair. The data type is reduced if the double `[[]]` pair is used. e.g.
```
> x <- list(foo = 1, bar = 2)
> x["foo"] |> str()
List of 1
 $ foo: num 1
> x[["foo"]] |> str()
 num 1
``

Reply:
If `headers` is a character vector instead of a list then the named element lookup will fail. e.g.
```
> x <- setNames(1:3, letters[1:3])
> x
a b c 
1 2 3 
> x[["d"]]
Error in x[["d"]] : subscript out of bounds
> x["d"]
<NA> 
  NA
```
I recall headers being a list so it avoids that hiccup. Well it is in `flipAI`, not sure in this repo.